Lib/distutils/tests/test_ccompiler.py

```diff
@@ -24,6 +24,33 @@ def library_option(self, lib):
 
 class CCompilerTestCase(support.EnvironGuard, unittest.TestCase):
 
+    def test_set_executables(self):
+        class MyCCompiler(CCompiler):
+            executables = {'compiler': '', 'compiler_cxx': '', 'linker': ''}
+
+        compiler = MyCCompiler()
+
+        # set executable as list
+        compiler.set_executables(compiler=['env', 'OMPI_MPICC=clang', 'mpicc'])
+        self.assertEqual(len(compiler.compiler), 3)
```

Comment:
Checking the length became redundant with the following check and so should be removed.